### ISSUE TITLE
Nomos 2.4.0: GPLv3+ recognized as GPL-2.0+

### ISSUE THREAD

**Issue by yaobinshi** [fossology/fossology#260]
---

Author Name: **armijn hemel** 
Original Redmine Issue: 6441, http://www.fossology.org/issues/6441
Original Date: 2014/01/21
Original Assignee: Bob Gobeille

---

the attached file is recognized as GPL-2.0+ while it really is GPLv3+

**Comment by yaobinshi** [comment]
---

Original Redmine Comment
Author Name: **larry shi** 
Original Date: 2014/01/22 08:59:01 +0000

---

besides GPL-3.0+, also include GPL-2.0+ in this file below?

143	
 * dug this out of the trashcan of the midgard project, lets see if it works for us.
144	
 * original code by Alexander Bokovoy <bokovoy@avilink.ne> distributed under GPL V2 or later

might be GPL-2.0+ is better? 	or GPL-3.0+ and GPL-2.0+? what do you think?

**Comment by yaobinshi** [comment]
---

Original Redmine Comment
Author Name: **Bob Gobeille** 
Original Date: 2014/02/10 18:51:04 +0000

---

I would call it GPL-3.0+ and GPL-2.0+ since it has both.

**Comment by yaobinshi** [comment]
---

Original Redmine Comment
Author Name: **larry shi** 
Original Date: 2014/02/18 09:19:53 +0000

---

it is difficult for nomos to report GPL-2.0 and GPL-3.0 together, that means have to do lots of code change, also will slow down the performance.

**Comment by yaobinshi** [comment]
updated by larry shi 10 months ago
Status changed from In Progress to Feedback
Assignee changed from larry shi to Bob Gobeille
according to the current logic of nomos, GPL-2.0 and GPL-3.0 are exclusive, can not report them together. 
only if change the logic of nomos, that means have to do lots of code change, have to refactor nomos, side effect is not under control, also will slow down the performance.

what do you think?

#5 Updated by Bob Gobeille 10 months ago
Status changed from Feedback to New
Priority changed from High to Normal
Target version changed from 2.5.0 to 2.6.0
Rank changed from 99 to 40
This is a nomos defect and since it involves a major change I'm pushing this to a future release. Nomos does need refactoring and this just documents one of the reasons. This is a serious error.

**Comment by pguttmann** [comment]
Attachment - http://www.fossology.org/attachments/4485/serv_rssclient.c

**Comment by yaobinshi** [comment]
will return GPL-2.0+&GPL-3.0+

**Comment by bobgob** [comment]
Please add test case to nomos/agent_tests/testdata/NomosTestfiles

**Comment by bobgob** [comment]
Why do you want to report a single  "GPL-2.0+&GPL-3.0+" instead of two licenses ("GPL-2.0+", "GPL-3.0+")?   In other cases we report individual licenses in the file and don't aggregate them with "&".  For example, if there were MIT, BSD, and GPL in a single file we report all the licenses and not "MIT&BSD&GPL"

**Comment by yaobinshi** [comment]
Good question, explained long time ago, for nomos, it is almost impossible to return all common licenses in one file, there are always if else clauses.
for this issue, hope being able to  find a better solution,

**Comment by mcjaeger** [comment]
tested as of today, see screenshot. Well it is not too bad.

![2015-08-28 20_47_51-change concluded license](https://cloud.githubusercontent.com/assets/3418977/9554436/278fbb74-4dc6-11e5-83fd-0cc0c6fd3076.png)

**Comment by bobgob** [comment]
Code has both GPL-3.0+ and GPL-2.0+

**Comment by mcjaeger** [comment]
Indeed, code has both in it and ... the ticket could be closed, but in fact it is bad for now.

We have a finding which says "GPL-2.0+&GPL-3.0+" which is completely not SPDX style and therefore should be moved out. Renaming the issue.

<img width="1045" alt="screen shot 2018-07-10 at 12 03 37" src="https://user-images.githubusercontent.com/3418977/42503577-54328f74-8439-11e8-8397-e17a30bac2e2.png">

Nomos should just detect two separate licenses.

Link to file is actually: 

http://archive15.fossology.org/attachments/4485/serv_rssclient.c

**Comment by GMishx** [comment]
Just tested now, seems to be working fine on latest master. But unfortunately highlight is not working on the file.

![2019-01-21 12_11_04-virtualbox](https://user-images.githubusercontent.com/18077542/51725876-917d6280-208a-11e9-8d92-c6233aa38980.png)

**Comment by shaheemazmalmmd** [comment]
Please re-open if the problem persists.
